packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -5882,7 +5901,57 @@ export default class Meeting extends StatelessWebexPlugin {
       throw error;
     }
 
-    await this.waitForMediaConnectionConnected();
+    try {
+      await this.waitForMediaConnectionConnected();
+
+      if (this.retriedWithTurnServer) {
+        Metrics.sendBehavioralMetric(
+          BEHAVIORAL_METRICS.RETRY_MEDIA_CONNECTION_WITH_TURN_SERVER_SUCCESS,
+          {
+            correlation_id: this.correlationId,
+          }
+        );
+      }
+    } catch (error) {
+      // If waitForMediaConnectionConnected() fails when we haven't done turn discovery (and the config allows for turn discovery to happen) then we
+      // attempt to establish a media connection again but this time using turn discovery. If we used turn discovery on the first pass we do not attempt
+      // connection again.
+
+      // @ts-ignore - config coming from registerPlugin
+      if (!this.turnServerUsed && this.turnDiscoverySkippedReason !== 'config') {
+        LoggerProxy.logger.error(
+          `${LOG_HEADER} error waiting for media to connect on UDP/TCP, retrying using TURN server, `,
+          error
+        );
+        this.retriedWithTurnServer = true;
```

Comment:
I think we need to reset it to false at the beginning of addMedia() - in case someone calls addMedia() again

Reply:
Reset at the beginning of addMedia